packages/e2e-tests-playwright/specs/widgets/customizing-widgets.spec.js

```diff
@@ -0,0 +1,656 @@
+		await editParagraphWidget.click();
+		await expect( firstParagraphBlock ).toBeFocused();
+
+		const headingWidget = previewFrame.locator(
+			'.widget:has-text("First Heading")'
+		);
+
+		const editHeadingWidget = headingWidget.locator(
+			'role=button[name="Click to edit this widget."i]'
+		);
+		await editHeadingWidget.click();
+
+		const headingBlock = page.locator(
+			'role=document[name="Block: Heading"i] >> text="First Heading"'
+		);
+		await expect( headingBlock ).toBeFocused();
+	} );
+
+	test( 'should clear block selection', async ( {
+		page,
+		testUtils,
+		widgetsCustomizerPage,
+	} ) => {
+		await widgetsCustomizerPage.expandWidgetArea( 'Footer #1' );
+
+		const paragraphBlock = await widgetsCustomizerPage.addBlock(
+			'Paragraph'
+		);
+		await page.keyboard.type( 'First Paragraph' );
+		await testUtils.showBlockToolbar();
+
+		const blockToolbar = page.locator(
+			'role=toolbar[name="Block tools"i]'
+		);
+
+		// Expect clicking on the section title should clear the selection.
+		{
+			await page.click(
+				'role=heading[name="Customizing ▸ Widgets Footer #1"i][level=3]'
+			);
+			await expect( blockToolbar ).not.toBeVisible();
+
+			await paragraphBlock.focus();
+			await testUtils.showBlockToolbar();
+		}
+
+		// Expect clicking on the preview iframe should clear the selection.
+		{
+			await page.click( '#customize-preview' );
+			await expect( blockToolbar ).not.toBeVisible();
+
+			await paragraphBlock.focus();
+			await testUtils.showBlockToolbar();
+		}
+
+		// Expect clicking on the empty space at the end of the editor
+		// should clear the selection.
+		{
+			const editorContainer = page.locator(
+				'#customize-control-sidebars_widgets-sidebar-1'
+			);
+			const { x, y, width, height } = await editorContainer.boundingBox();
+			// Simulate Clicking on the empty space at the end of the editor.
+			await page.mouse.click( x + width / 2, y + height + 10 );
+			await expect( blockToolbar ).not.toBeVisible();
+		}
+	} );
+
+	test( 'should handle legacy widgets', async ( {
+		page,
+		testUtils,
+		widgetsCustomizerPage,
+	} ) => {
+		await widgetsCustomizerPage.expandWidgetArea( 'Footer #1' );
+
+		const legacyWidgetBlock = await widgetsCustomizerPage.addBlock(
+			'Legacy Widget'
+		);
+		await page
+			.locator(
+				'role=combobox[name="Select a legacy widget to display:"i]'
+			)
+			.selectOption( 'test_widget' );
+
+		await expect(
+			legacyWidgetBlock.locator(
+				'role=heading[name="Test Widget"][level=3]'
+			)
+		).toBeVisible();
+
+		let titleInput = legacyWidgetBlock.locator(
+			'role=textbox[name="Title:"i]'
+		);
+
+		await titleInput.type( 'Hello Title' );
+
+		// Unfocus the current legacy widget.
+		await page.keyboard.press( 'Tab' );
+
+		// Disable reason: Sometimes the preview just doesn't fully load,
+		// it's the only way I know for now to ensure that the iframe is ready.
+		// eslint-disable-next-line no-restricted-syntax
+		await page.waitForTimeout( 2000 );
```

Comment:
```js
		const legacyWidgetPreviewFrame = page.frameLocator(
			'iframe[title="Legacy Widget Preview"]'
		);
    await legacyWidgetPreviewFrame.waitFor();
```
I'd try to nuke the waitForTimeout and `waitForPreviewIframe` method entirely.

Reply:
Yeah makes sense, now that we're using locator which is lazy, I guess we don't have to do this hack anymore.